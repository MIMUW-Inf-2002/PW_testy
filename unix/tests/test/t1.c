#include <signal.h>
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <unistd.h>

static int count = 0;

void sig_handler(int signo) { fprintf(stderr, "singint # %d\n", count++); }

int cur_time() {
  struct timespec ts;
  clock_gettime(CLOCK_MONOTONIC, &ts);
  return ts.tv_sec * 1000 + ts.tv_nsec / 1000000;
}

int main() {
  signal(SIGINT, sig_handler);
  fprintf(stderr, "Nice to meet you\n");

  int end = 3000 + cur_time();

  while (cur_time() < end) {
    int x = cur_time();
    usleep(end - x);
  }

  fprintf(stdout, "Goodbye\n");
  return 0;
}
