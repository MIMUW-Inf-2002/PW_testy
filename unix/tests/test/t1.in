run ./t1
out 0
sleep 1000
run ./t1
sleep 100
err 1
kill 0
kill 1
sleep 100
err 0
out 0
err 0
out 1
err 1
kill 0
kill 1
sleep 100
sleep 2000
out 0
err 0
out 1
err 1
quit
out 1
err 1
