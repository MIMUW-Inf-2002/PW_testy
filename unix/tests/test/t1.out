Task 0 started: pid 8287.
Task 0 stdout: ''.
Task 1 started: pid 8291.
Task 1 stderr: 'Nice to meet you'.
Task 0 stderr: 'singint # 0'.
Task 0 stdout: ''.
Task 0 stderr: 'singint # 0'.
Task 1 stdout: ''.
Task 1 stderr: 'singint # 0'.
Task 0 ended: status 0.
Task 0 stdout: 'Goodbye'.
Task 0 stderr: 'singint # 1'.
Task 1 stdout: ''.
Task 1 stderr: 'singint # 1'.
Task 1 ended: signalled.
